lib/ssh_scan/authenticator.rb

```diff
@@ -7,7 +7,7 @@ def initialize(config)
     end
 
     def self.from_config_file(config_file)
-      opts = YAML.load_file(config_file)
+      opts = SSHScan::ConfigLoader.load(config_file, SSHScan::Authenticator)
```

Comment:
I think you mean `self.class`. `self` will be an instance, while I want the Class type.